timeseer/podman/roles/timeseer_data_service/tasks/main.yml

```diff
@@ -15,27 +15,40 @@
     timeseer_data_service_storage_mount: "{{ timeseer_data_service_dir }}"
   when: timeseer_data_service_podman_volume | default('') | trim == '' and timeseer_data_service_enable
 
-- name: Run ts-data-service as a container
-  containers.podman.podman_container:
-    name: timeseer-data-service
-    image: "{{ timeseer_data_service_image }}"
-    command: "/usr/bin/timeseer-data-service"
-    state: started
-    pull: "{{ timeseer_data_service_image_pull }}"
-    restart_policy: unless-stopped
-    network: "{{ timeseer_data_service_network }}"
-    volumes: "{{ volume_list }}"
-    log_driver: journald
-    log_options:
-      tag: "{{ timeseer_data_service_journal_tag }}"
-    env:
-      RUST_LOG: info
-    tty: true
-    ports: "{{ timeseer_data_service_ports }}"
+- name: Define timeseer-data-service volume list
+  ansible.builtin.set_fact:
+    data_service_volume_list: "{{ ds_base_volumes + (ds_custom_toml_volume if timeseer_data_service_config_dir | default('') | trim != '' else []) }}"
   vars:
-    base_volumes:
+    ds_base_volumes:
       - "{{ timeseer_data_service_storage_mount }}:/usr/src/app/data-service"
-    custom_toml_volume:
+    ds_custom_toml_volume:
       - "{{ timeseer_data_service_config_dir }}:/usr/src/app/Timeseer-data-service.toml"
-    volume_list: "{{ base_volumes + (custom_toml_volume if timeseer_data_service_config_dir | default('') | trim != '' else []) }}"
+
+- name: Prepend 'systemd-' to network
+  ansible.builtin.set_fact:
+    timeseer_data_service_network: "systemd-{{ timeseer_data_service_network }}"
+
+- name: Template configuration files
+  ansible.builtin.template:
+    src: "{{ item }}"
+    dest: "/etc/containers/systemd/timeseer-data-service.container"
+    mode: "0644"
+  loop:
+    - "timeseer-data-service-container.quadlet"
+  register: timeseer_data_service_container
+  notify:
+    - Restart timeseer-data-service service
+  when: timeseer_data_service_enable
```

Comment:
It's cleaner and less error-prone to either have a block in `main.yml` with this condition or have a conditional include.

Reply:
The idea was not that there is one task in an include, but to have all tasks in this file moved to that separate file. They all should only run when the data service is enabled. That's why the one block here is also a valid option.